prototype/mobile_endpoint/backends/mongo/dao.py

```diff
@@ -1,58 +1,70 @@
-from collections import defaultdict
+from uuid import UUID
+from mongoengine import DoesNotExist
 from mobile_endpoint.dao import AbsctractDao, to_generic
-from mobile_endpoint.backends.mongo.models import MongoForm, MongoCase
-from couchdbkit import ResourceNotFound
+from mobile_endpoint.backends.mongo.models import MongoForm, MongoCase, \
+    MongoSynclog
+from mobile_endpoint.exceptions import NotFound
 from mobile_endpoint.utils import get_with_lock
 
 
 class MongoDao(AbsctractDao):
 
     def commit_atomic_submission(self, xform, case_result):
-
-        docs_by_collection = defaultdict(list)
+        # TODO: Save all in one bulk operation
+        # TODO: Are the forms, cases, and synclogs all supposed to be all or nothing?
```

Comment:
i think this is fine for now. main thing is to get performance benchmarks not have perfectly ACID code